lib/Scheduling/Utilities.cpp

```diff
@@ -154,5 +159,4 @@ void scheduling::dumpAsDOT(Problem &prob, StringRef fileName) {
 
   os.unindent();
   os << "}\n";
-  out.close();
```

Comment:
It might be nice to `close` here. I believe the `close` call is technically missing from the `printInstance` function in the pass, and it would be nice to not have to remember to use it.

Reply:
Hmm, my assumption was that neither the pass nor the utility own the stream they get as an argument, so it felt wrong to me to close the stream after dumping an instance. Also, `raw_ostream` doesn't even offer a `close()` method (which seems to be only available on the file-centric subclasses).

Reply:
Oh I see, then this makes sense. Yeah leaving it up to the caller to close (when possible) sounds best.